### ISSUE TITLE
onUploadPacketsReceived not fired

### ISSUE THREAD

**Issue by tiwiz** [akinaru/speed-test-lib#12]
Often, when reaching a percent of 100, the method is not called.

This is a "non-issue", because checking for the percentage can solve it, but I wanted to let you know anyways

**Comment by akinaru** [comment]
Maybe it is the same as issue #9 . Do you call `startUpload` in one of the callback ? If it is the case, it's a thread problem that will be fixed in next release, for now you can run it in a new thread :
```
new Thread(new Runnable() {
    @Override
    public void run() {
        speedTestSocket.startUpload(SPEED_TEST_SERVER_HOST, SPEED_TEST_SERVER_PORT, SPEED_TEST_SERVER_URI_UL,
                FILE_SIZE);
    }
}).start();
```

**Comment by tiwiz** [comment]
Thanks for the prompt answer! I already did it in a new Observable and everything is fine :-)

**Comment by akinaru** [comment]
The referenced bug has been fixed in v`1.21`
